logstreams/src/main/java/io/zeebe/distributedlog/impl/DefaultDistributedLogstreamService.java

```diff
@@ -263,72 +245,58 @@ public void backup(BackupOutput backupOutput) {
   @Override
   public void restore(BackupInput backupInput) {
     final long backupPosition = backupInput.readLong();
-    restore(backupPosition);
-
-    logger.debug("Restored local log to position {}", lastPosition);
-    currentLeader = backupInput.readString();
-    currentLeaderTerm = backupInput.readLong();
-  }
 
-  public void restore(long backupPosition) {
     if (lastPosition < backupPosition) {
       LogstreamConfig.startRestore(localMemberId, partitionId);
+      final RestoreFactory restoreFactory = LogstreamConfig.getRestoreFactory(localMemberId);
+
+      final ThreadContext restoreThreadContext =
```

Comment:
Just a tiny improvement, `AutoCloseable` things can be wrapped in `try-with-resources`:

```java
try(ThreadContext restoreThreadContext = new SingleThreadContext(String.format("log-restore-%d", partitionId))) {
...
}
```

Which gets closed safely once the try block exits. Though that will create even more indentation, maybe we can split the rest into smaller methods to make it more readable? :thinking: